qiskit/transpiler/passes/calibration/builders.py

```diff
@@ -170,14 +170,14 @@ def rescale_cr_inst(instruction: Play, theta: float, sample_mult: int = 16) -> P
 
             if target_area > gaussian_area:
                 width = (target_area - gaussian_area) / abs(amp)
-                duration = math.ceil((width + n_sigmas * sigma) / sample_mult) * sample_mult
+                duration = round((width + n_sigmas * sigma) / sample_mult) * sample_mult
```

Comment:
I think this is correct fix. Could you please write unittest?

Test file is here. Perhaps we should move the file under `test.python.transpiler` but not the scope of this PR.
https://github.com/Qiskit/qiskit-terra/blob/main/test/python/pulse/test_calibrationbuilder.py

Reply:
Do you have a suggestion for how this should be tested? This bug is very subtle and does not seem to be triggered by any of the mock devices. Apparently whenever this line is called, any potential difference between `math.ceil` and `round` is due to floating point error.

Reply:
Since `rescale_cr_inst` is fortunately a static method, you can directly test it without using actual backend object. Just create a fake `Play` instruction with problematic fake CR pulse instance and run the method with it. You can check the duration of returned pulse.

Reply:
Thanks, I've added the test.